lib/parser/source/range.rb

```diff
@@ -86,6 +86,27 @@ def line
 
       alias_method :first_line, :line
 
+      ##
+      # The range of line numbers spanned by the range.
+      # The range will be exclusive or not depending on the keyword argument;
+      # default is inclusive.
+      #
+      # Assume that `def_node` is the Node corresponding to the following example:
+      #
+      #    def foo
+      #      do_something
+      #      42
+      #    end
+      #
+      # Then:
+      #
+      #    def_node.loc.end.line_span                           # => 4..4
+      #    def_node.loc.expression.line_span(exclude_end: true) # => 1...4
+      #
+      def line_span(exclude_end: false)
```

Comment:
It looks a bit confusing to me that it returns different ranges because of this boolean argument. I'd expect it to return:

+ `1..4` for `exclude_end = false`
+ `1...5` for `exclude_end = true`

What's the reason to add this method? Are there examples in rubocop that you could share? `exclude_end` doesn't "exclude" the `end` keyword in general (because there can be (a) single line method def and (b) not all statements have `end` keyword, this method doesn't make sense for oneliners like lvar assign)

Reply:
Right, this can be confusing. I was not referring to `end` keyword, or `location.end`, but to `::Range#exclude_end?`. 

For `1...4`, the thinking is to exclude `4` as the `def` doesnt' include that line's ending, but maybe it's just not clean enough overall.

For RuboCop, the typical use is to deal with comments; whenever we replace a particular range, then we need to take care to swallow any comments within that range (`exclude_end == true`), or if we want to know if it that range is commented (`exclude_end == false`).

Maybe it's best to keep it within RuboCop then?

Reply:
Yes, it seems to be very Rubocop -specific

Reply:
Ok, thanks for your time 👍